lib/clusterManager.js

```diff
@@ -189,7 +194,64 @@ if (cluster.isMaster) {
     this.overrideFileConfig = override;
   }
 
-   ClusterManager.prototype.notifyWorkersForAddingPlugin = function(resultHandler, indexInCluster) {
+  /* Having gotten here, we are already at the cluster, but it's safer to validate anyway
+    i. e. "if (this.storage) ..." since only the master cluster has storage */
+  ClusterManager.prototype.getStorage = function(resultHandler) {
+    if (this.storage) {
+      return resultHandler(this.storage);
+    } else {
+      return resultHandler(zluxUtil.DataserviceStorage());
+    }
+  }
+
+  ClusterManager.prototype.setStorageValue = function(id, value, resultHandler) {
+    if (this.storage) {
+      this.storage.setStorageValue(id, value);
+      return resultHandler(true);
+    } else {
+      clusterLogger.warn("ZWED0160W - Master cluster not found to save storage for id: " + id + ", value: " + value);
+      return resultHandler(false);
+    }
+  }
+
+  ClusterManager.prototype.setStorage = function(dict, resultHandler) {
+    if (this.storage) {
+      this.storage.setStorage(dict);
+      return resultHandler(true);
+    } else {
+      clusterLogger.warn("ZWED0161W - Master cluster not found to save storage");
+      return resultHandler(false);
+    }
+  }
+
+  ClusterManager.prototype.addStorage = function(dict, resultHandler) {
+    if (this.storage) {
+      const mergedObjects = recursiveMerge(this.storage.getStorage(), dict);
+      this.storage.setStorage(mergedObjects);
+      return resultHandler(true);
+    } else {
+      clusterLogger.warn("ZWED0162W - Master cluster not found to add storage");
+      return resultHandler(false);
+    }
+  }
+
+  ClusterManager.prototype.getStorageValue = function(id, resultHandler) {
+    if (this.storage) {
+      return resultHandler(this.storage.getObject(id));
+    } else {
+      return resultHandler(null);
+    }
+  }
+
+  ClusterManager.prototype.deleteStorageValue = function(id) {
```

Comment:
1. Good catch! @1000TurquoisePogs Is "deleteStorageValue"
2. Solely because I forgot to add a callback. I will make it async, so it has a callback, just as the others do, as we will need our users to be informed when these cluster operations are finished